pkg/file/pipeline/hashtrie/hashtrie.go

```diff
@@ -21,20 +25,29 @@ var (
 const maxLevel = 8
 
 type hashTrieWriter struct {
+	ctx                    context.Context // context for put function of dispersed replica chunks
 	refSize                int
 	cursors                []int  // level cursors, key is level. level 0 is data level holds how many chunks were processed. Intermediate higher levels will always have LOWER cursor values.
 	buffer                 []byte // keeps intermediate level data
 	full                   bool   // indicates whether the trie is full. currently we support (128^7)*4096 = 2305843009213693952 bytes
 	pipelineFn             pipeline.PipelineFunc
 	rParams                redundancy.IParams
 	parityChunkFn          redundancy.ParityChunkCallback
-	chunkCounters          []uint8 // counts the chunk references in intermediate chunks. key is the chunk level.
-	effectiveChunkCounters []uint8 // counts the effective  chunk references in intermediate chunks. key is the chunk level.
-	maxChildrenChunks      uint8   // maximum number of chunk references in intermediate chunks.
+	chunkCounters          []uint8        // counts the chunk references in intermediate chunks. key is the chunk level.
+	effectiveChunkCounters []uint8        // counts the effective  chunk references in intermediate chunks. key is the chunk level.
+	maxChildrenChunks      uint8          // maximum number of chunk references in intermediate chunks.
+	replicaPutter          storage.Putter // putter to save dispersed replicas of the root chunk
```

Comment:
if only used for replicas then this should be right away initialised  with `replicas.NewPutter`